File: parseDesiderata.js
const XLSX = require('xlsx');
const path = require('path');
const fs = require('fs');

// Liste des fichiers XLSX à traiter
const xlsxFiles = [
  'FRANCON  T.XLSX',
  'GOUMIDI  T.XLSX',
  'KHERFI  T.XLSX',
  'LEDIAGON  T.XLSX',
  'SINANIAN  T.XLSX',
  'ZIAN  T.XLSX',
  'ZWANEVELD  T.XLSX'
];

const desideratasFolder = path.join(__dirname, 'Desideratas papiers');

// Mapping des créneaux
const creneauxMapping = {
  '1er QUART': 'QUART_1',
  '2ème QUART': 'QUART_2',
  'RENFORT': 'RENFORT_1',
  '3ème QUART': 'QUART_3',
  'RENFORT 2': 'RENFORT_2',
  '4ème QUART': 'QUART_4'
};

// Fonction pour parser un fichier XLSX
function parseXLSX(filePath) {
  try {
    const workbook = XLSX.readFile(filePath);
    const sheetName = workbook.SheetNames[0];
    const sheet = workbook.Sheets[sheetName];

    // Convertir en JSON
    const data = XLSX.utils.sheet_to_json(sheet, { header: 1, raw: false, defval: '' });

    console.log(`\n=== ${path.basename(filePath)} ===`);
    console.log(`Nombre de lignes: ${data.length}`);
    console.log(`Première ligne (en-têtes):`, data[0]);
    console.log(`Deuxième ligne:`, data[1]);
    console.log(`Troisième ligne:`, data[2]);

    return {
      fileName: path.basename(filePath),
      data: data,
      headers: data[0] || []
    };
  } catch (error) {
    console.error(`Erreur lors de la lecture de ${filePath}:`, error.message);
    return null;
  }
}

// Parser tous les fichiers
const results = [];
for (const file of xlsxFiles) {
  const filePath = path.join(desideratasFolder, file);
  if (fs.existsSync(filePath)) {
    const result = parseXLSX(filePath);
    if (result) {
      results.push(result);
    }
  } else {
    console.log(`Fichier non trouvé: ${filePath}`);
  }
}

// Sauvegarder les résultats
fs.writeFileSync(
  path.join(__dirname, 'desiderata-parsed.json'),
  JSON.stringify(results, null, 2)
);

console.log('\n✓ Résultats sauvegardés dans desiderata-parsed.json');

File: parse_desiderata.py
import openpyxl
import json
import os
from datetime import datetime

# Liste des fichiers XLSX à traiter
xlsx_files = [
    'FRANCON  T.XLSX',
    'GOUMIDI  T.XLSX',
    'KHERFI  T.XLSX',
    'LEDIAGON  T.XLSX',
    'SINANIAN  T.XLSX',
    'ZIAN  T.XLSX',
    'ZWANEVELD  T.XLSX'
]

desideratas_folder = os.path.join(os.path.dirname(__file__), 'Desideratas papiers')

# Mapping des créneaux
creneaux_mapping = {
    '1er QUART': 'QUART_1',
    '2ème QUART': 'QUART_2',
    '2EME QUART': 'QUART_2',
    'RENFORT': 'RENFORT_1',
    '3ème QUART': 'QUART_3',
    '3EME QUART': 'QUART_3',
    'RENFORT 2': 'RENFORT_2',
    '4ème QUART': 'QUART_4',
    '4EME QUART': 'QUART_4'
}

# Fonction pour parser un fichier XLSX
def parse_xlsx(file_path):
    try:
        wb = openpyxl.load_workbook(file_path, data_only=True)
        sheet = wb.active

        data = []
        for row in sheet.iter_rows(values_only=True):
            data.append(list(row))

        print(f"\n=== {os.path.basename(file_path)} ===")
        print(f"Nombre de lignes: {len(data)}")
        print(f"Première ligne (en-têtes): {data[0] if len(data) > 0 else 'N/A'}")
        if len(data) > 1:
            print(f"Deuxième ligne: {data[1]}")
        if len(data) > 2:
            print(f"Troisième ligne: {data[2]}")

        return {
            'fileName': os.path.basename(file_path),
            'data': data,
            'headers': data[0] if len(data) > 0 else []
        }
    except Exception as e:
        print(f"Erreur lors de la lecture de {file_path}: {str(e)}")
        return None

# Parser tous les fichiers
results = []
for file in xlsx_files:
    file_path = os.path.join(desideratas_folder, file)
    if os.path.exists(file_path):
        result = parse_xlsx(file_path)
        if result:
            results.append(result)
    else:
        print(f"Fichier non trouvé: {file_path}")

# Sauvegarder les résultats
with open(os.path.join(os.path.dirname(__file__), 'desiderata-parsed.json'), 'w', encoding='utf-8') as f:
    json.dump(results, f, ensure_ascii=False, indent=2, default=str)

print('\n✓ Résultats sauvegardés dans desiderata-parsed.json')
